boa_engine/src/builtins/eval/mod.rs

```diff
@@ -83,12 +83,10 @@ impl Eval {
 
         // Parse the script body (11.a - 11.d)
         // TODO: Implement errors for 11.e - 11.h
-        let parse_result = if strict {
-            context.parse_strict(x.as_bytes())
-        } else {
-            context.parse(x.as_bytes())
-        };
-        let body = match parse_result.map_err(|e| e.to_string()) {
+        let body = match context
+            .parse_eval(x.as_bytes(), direct, strict)
+            .map_err(|e| e.to_string())
```

Comment:
Could we avoid the closure here, and directly point to the `to_string` function?